worlds/kh2/__init__.py

```diff
@@ -23,85 +24,102 @@ class KingdomHearts2Web(WebWorld):
     )]
 
 
-# noinspection PyUnresolvedReferences
 class KH2World(World):
     """
     Kingdom Hearts II is an action role-playing game developed and published by Square Enix and released in 2005.
     It is the sequel to Kingdom Hearts and Kingdom Hearts: Chain of Memories, and like the two previous games,
     focuses on Sora and his friends' continued battle against the Darkness.
     """
-    game: str = "Kingdom Hearts 2"
+    game = "Kingdom Hearts 2"
     web = KingdomHearts2Web()
-    data_version = 1
-    required_client_version = (0, 4, 0)
+    data_version = 2
+
+    required_client_version = (0, 4, 2)
     option_definitions = KH2_Options
-    item_name_to_id = {name: data.code for name, data in item_dictionary_table.items()}
-    location_name_to_id = {item_name: data.code for item_name, data in all_locations.items() if data.code}
+    item_name_to_id = {item: item_id
+                       for item_id, item in enumerate(item_dictionary_table.keys(), 0x130000)}
+    location_name_to_id = {item: location
+                           for location, item in enumerate(all_locations.keys(), 0x130000)}
     item_name_groups = item_groups
 
     def __init__(self, multiworld: "MultiWorld", player: int):
         super().__init__(multiworld, player)
+        self.keyblade_ability_pool = None
+        self.goofy_get_bonus_abilities = None
+        self.goofy_weapon_abilities = None
+        self.donald_get_bonus_abilities = None
+        self.donald_weapon_abilities = None
+
         self.valid_abilities = None
         self.visitlocking_dict = None
         self.plando_locations = None
-        self.luckyemblemamount = None
-        self.luckyemblemrequired = None
-        self.BountiesRequired = None
-        self.BountiesAmount = None
+        self.lucky_emblem_amount = None
+        self.lucky_emblem_required = None
+        self.bounties_required = None
+        self.bounties_amount = None
         self.hitlist = None
-        self.LocalItems = {}
-        self.RandomSuperBoss = list()
+        self.random_super_boss_list = list()
         self.filler_items = list()
         self.item_quantity_dict = {}
-        self.donald_ability_pool = list()
-        self.goofy_ability_pool = list()
+
         self.sora_keyblade_ability_pool = list()
         self.keyblade_slot_copy = list(Locations.Keyblade_Slots.keys())
         self.keyblade_slot_copy.remove(LocationName.KingdomKeySlot)
-        self.totalLocations = len(all_locations.items())
+        self.total_locations = len(all_locations.keys())
         self.growth_list = list()
         for x in range(4):
             self.growth_list.extend(Movement_Table.keys())
-        self.slotDataDuping = set()
-        self.localItems = dict()
+        self.slot_data_duping = set()
+        self.local_items = dict()
 
     def fill_slot_data(self) -> dict:
+        # localItems filling done here for the unit test.
```

Comment:
the localItems variable is used to make sure that the player has the correct amount of items. The main use for this is if the player dies the l open the chests and give them what was there. The reason it was moved to fill_slot_data was sometimes it ended up being empty but that has since been fixed.